[FEAT]: [BE] 관리자 주문 조회 기능 추가 등

loadingKKamo21 opened the pull request:
- 관리자 권한 대상 접근 가능 URL 구현
  · 모든 주문 목록 조회(/api/v1/admin/orders) 추가
- OrderMessageConstant 메시지 1개 추가
  · "주문 목록 조회 성공" 추가<!-- 제목 : convention: 기능명#issue 번호
  ex) [FEAT] : pull request template #17-->

## 📝Part
- [x] BE
- [ ] FE
- [ ] Infra

  <br/>

## ✔️PR Type
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 주요 코드 리펙토링 (성능 최적화 등)
- [x] 간단 코드 리펙토링 (주석, 코드 컨벤션, 오타 수정 등)
- [ ] 기타 (기타 사항 기입)

  <br/>

## ✔️PR Checklist
- [x] 커밋 메세지를 컨벤션에 맞게 잘 적용 하였나요?
- [x] 테스트 코드 작성 / 단위 테스트 or 통합 테스트 진행 하셨나요?

  <br/>

## 🔎 작업 내용

<!--#{Issue번호} + Enter
ex) #17
이슈와 PR 연결을 위해 사용합니다!-->

- 관리자의 모든 주문 목록 조회 구현
	- 테스트 코드 작성
- 시큐리티 적용에 따른 인증 필요 API 수정
	- 주문(Order) 저장, 조회, 취소에 적용
	- 주문 관련 에러 코드 1개 추가
	- 관련 테스트 코드 추가 및 수정(OrderRepository, OrderService, OrderController)
- 더미 데이터 생성 클래스(BaseInitData) 수정
	- 회원 엔티티 생성 시 비밀번호 암호화 적용

  <br/>

## 📈이미지 첨부

loadingKKamo21 committed:
[FEAT]: [BE] AdminController 추가

- 관리자 권한 대상 접근 가능 URL 구현
  · 모든 주문 목록 조회(/api/v1/admin/orders) 추가
- OrderMessageConstant 메시지 1개 추가
  · "주문 목록 조회 성공" 추가

loadingKKamo21 committed:
[TEST]: [BE] AdminController 테스트 추가

loadingKKamo21 committed:
[FIX]: [BE] 시큐리티 URL 설정 추가

- 지정된 URL이 아닌 URL에 대한 허용/비허용 설정 추가

loadingKKamo21 committed:
[REFACTOR]: [BE] AdminController 테스트 미사용 필드 제거 및 인증 객체 이름 변경

DayDreaam reviewed (approved):
확인 완료했습니다! 찬우님 고생 많으셨습니다!

KangBaekGwa commented on backend/src/main/java/com/app/backend/global/security/SecurityConfig.java:
```diff
@@ -22,23 +22,24 @@ public class SecurityConfig {
+                                           authorizeRequests
+                                                   .requestMatchers("/h2-console/**").permitAll()
+                                                   .requestMatchers("/api/*/signup").permitAll()
+                                                   .requestMatchers("/api/v1/users/**").permitAll()
```
users 에 하위 리소스 붙어있는거 대부분, 인증 필요해 보입니다.
- 회원탈퇴 : /users/{userId} [Delete]
- 회원 정보 조회 : /users/{userId} [Get]
- 회원 정보 수정 : /users/{userId} [Patch]
- ... 등등

--- 

다른 path 들도 점검 부탁드립니다.

---

혹시, user 파트 개발자가 진행하기로 했다면 해당 코멘트 resolve 해주세요 ㅎㅎ..

sungyeong98 replied:
그 부분은 아직 처리 중이어서 제가 나중에 제거하겠습니다.

KangBaekGwa replied:
넵 확인했습니다~!

KangBaekGwa commented on backend/src/main/java/com/app/backend/domain/user/controller/AdminController.java:
```diff
@@ -0,0 +1,50 @@
+    @GetMapping("/orders")
+    public RsData<List<OrderResponse>> getAllOrders(@AuthenticationPrincipal UserDetails userDetails) {
+        System.out.println("userDetails = " + userDetails);
+        if (!userDetails.getAuthorities().stream()
+                        .anyMatch(authority -> authority.getAuthority().equals("ROLE_ADMIN")))   //관리자 권한 확인
```
이부분, securityConfig 에서, 해당 url 매핑 hasRole 로 처리하면 안먹나요? 궁금합니다.
```
.requestMatchers(HttpMethod.GET, "/api/v1/orders").hasRole("ADMIN")
```

---

가능하면, 그렇게 진행한다면 사용자에 대한 접근 권한은 모두 security 에서 자체 처리가 되므로, 책임분리 되어 유지보수성 좋아질 것 같습니다.

loadingKKamo21 replied:
맞습니다. 시큐리티 설정 merge되면 일괄 수정하면 될 것 같습니다.

loadingKKamo21 committed:
Merge branch 'refs/heads/main' into feat-admin-orders

# Conflicts:
#	backend/src/main/java/com/app/backend/global/security/SecurityConfig.java

loadingKKamo21 committed:
[FEAT]: [BE] 시큐리티 인증 활용을 위한 메서드 추가

- OrderRepository
 · findByIdAndCustomer_Id 추가
- OrderService
 · getOrderByIdAndUserId 추가
 · updateOrderStatusByUserId 추가
- OrderController
 · @AuthenticationPrincipal인증 객체 사용

loadingKKamo21 committed:
[TEST]: [BE] 변경 사항에 대한 테스트 추가 및 수정

loadingKKamo21 committed:
[FEAT]: [BE] 주문 관련 에러 코드(ErrorCode) 추가

loadingKKamo21 committed:
[TEST]: [BE] AdminController 테스트 수정

- 시큐리티 적용, 인증 실패 시 응답 상태 변경

loadingKKamo21 committed:
[FEAT]: [BE] 더미 데이터 생성 시 비밀번호 암호화 적용

- BaseInitData 내 임시 회원 데이터 비밀번호에 암호화 적용
